src/components/Modal/index.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './Modal.scss';
+import Button from '../Button';
+
+type Props = {
+  // Determines wether the modal is displayed or not. Note that the modal is only added to the DOM
+  // when its being displayed
+  isOpen: boolean,
+  // Set to true for modals that should be centered
+  center: boolean,
+  // This callback is called when the user wants to close the dialog.
+  onClose: () => void
+};
+
+const modalRoot = document.body;
+
+// Generic Modal Component
+// thanks to https://reactjs.org/docs/portals.html
+export default class Modal extends React.Component {
+  constructor(props: Props) {
+    super(props);
+    this.el = document.createElement('div');
+  }
+
+  componentDidMount() {
+    modalRoot.appendChild(this.el);
+    this._addBodyClass();
+  }
+
+  componentDidUpdate() {
+    this._addBodyClass();
```

Comment:
I'm wondering if we need a conditional check here to update the body class only when `isOpen` changes?

Reply:
good idea, that way we can avoid adding/removing the classes from the body unnecessarily.